src/planwise/util/ring.clj

```diff
@@ -34,3 +37,43 @@
     (println (str "\n\n=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-="))
     (clojure.pprint/pprint request)
     (handler request)))
+
+(def filtered-params #{"password"})
+
+(defn- filter-params
+  [params]
+  (into {} (map
+            (fn [[k v]]
+              [k (if (filtered-params (name k)) "[FILTERED]" v)])
+            params)))
+
+(defn wrap-log-request
+  "Middleware to log each HTTP request in a single line"
+  ([handler options]
+   (let [level (:level options :info)
+         exclude-uris (:exclude-uris options)]
+     (fn [request]
+       (let [uri (:uri request "")
+             params (:params request)]
+         (when-not (and (some? exclude-uris)
+                        (re-matches exclude-uris uri))
+           (log level "Starting request:"
+                (-> (:request-method request) name .toUpperCase)
+                (:uri request)
+                (or (:query-string request) ""))
```

Comment:
Maybe refactor these 3 lines into an auxiliary function, since they are repeated exactly in the function below?